Documentation/Blazorise.Docs/Pages/Blog/2024-05-17_HowToCreateSocialMediaShareButtons/Index.md

```diff
@@ -0,0 +1,205 @@
+```
+```cs|ShareButtonComponentCode
+@code 
+{
+  [Parameter]
+  public string TextColor { get; set; }
+
+  [Parameter]
+  public string BackgroundColor { get; set; }
+
+  [Parameter]
+  public string IconName { get; set; }
+
+  [Parameter, EditorRequired]
+  public string To { get; set; }
+
+  [Parameter, EditorRequired]
+  public RenderFragment ChildContent { get; set; }
+
+  [Parameter(CaptureUnmatchedValues = true)]
+  public Dictionary<string, object> AdditionalAttributes { get; set; } = [];
+}
+```
+
+## Let's break down the component
+
+### Button
+
+First thing's first, we have the Button component, notice that it is typed as **B**utton and not button that is because it is not an ordinary HTML button, it is a Blazorise button component! This means it can take parameters, to allow us to customize it further!
+
+> Just like other Blazorise components, this button is framework-agnostic, meaning you may use Bootstrap, TailwindCSS, or any other supported frameworks!
+
+### The parameters
+We have just enough parameters to allow for the exact customization necessary,
+
+Here is a breakdown of what each one does:
+- `TextColor` - The button's text color.
+- `BackgroundColor` - The button's background color, we will define custom brand colors soon.
+- `IconName` - The name of the icon displayed on the button, in this case we are using [FontAwesome](https://fontawesome.com/).
+- `To` - The link where the user is navigated to, will bind to the `href` attribute on a regular link.
+- `ChildContent` - The markup displayed inside the button. See [blazor-university](https://blazor-university.com/templating-components-with-renderfragements/).
+- `AdditionalAttributes` - Any additional attributes the user passes to the button. Will directly be applied to the underlying button component. See [blazor-university](https://blazor-university.com/components/capturing-unexpected-parameters/).
+
+---
+
+## Define the brand colors in `wwwroot/brands.css`
+Here are some colors, you may expand this further as you need
+
+```html|Brands
+.bg-snapchat {
+  background-color: #FFFC00 !important;
+}
+
+.bg-discord {
+  background-color: #5865F2 !important;
+}
+
+.bg-github {
+  background-color: #0D1117 !important;
+}
+
+.bg-x {
+  background-color: #000000 !important;
+}
+
+/* other brand colors. add the brands you need here! */
+```
+> The `!important` property, this is necessary as, by default the Bootstrap icons will have the `Color` property set to `primary`, this will shadow our custom background colors, so adding `!important` at the end of them will fix this.
+
+---
+
+## Include the `brands.css` in your app
+
+```html|IndexhtmlHeadSection
+<html>
+<head>
+  <link href="brands.css" rel="stylesheet" />
+  ...
+</head>
+</html>
+```
+
+---
+
+## Using the ShareButtons!
+
+Inside your page, add the freshly created buttons
+
+```html|ShareButtonUsage
+<ShareButton TextColor="white" BackgroundColor="x" IconName="fa-brands fa-x" To="https://twitter.com/intent/tweet">
```

Comment:
For the end-user I think it would be better to change the API a little

```
<ShareButton TextColor="white" Brand="x" To="https://twitter.com/intent/tweet">
    Share on
</ShareButton>
```

Notice I have removed `BackgroundColor`, and `IconName`. Maybe the component should handle it automatically based on a brand name?

Reply:
Good idea, I was thinking of implementing that, so that only the button is aware of the brands. 
Also, what if I made a brands enum?

Reply:
> Also, what if I made a brands enum?

It's not a bad idea.